### ISSUE TITLE
[HELP] Adding a large number of files via array.

### ISSUE THREAD

**Issue by Laurvin** [Touffy/client-zip#31]
This is neither a bug nor a feature but this seemed the best place to ask.

I want to save the 900+ list items on a page as separate files in a zip. It's easy enough to itterate them and put them in an array/object (most of my experience is with php so I always struggle with this). But I can't see how I can pass this to this line of code:

> wait downloadZip([intro, code]).blob()

Is there any way of doing this? The number of files will grow over time so I need a programmatic way of passing them all to the blob.

Thanks!

**Comment by ricky11** [comment]
We're doing exactly this.
1)registering the service worker
2)collecting all the urls in to an array
3) fetch the urls in the service worker
4) trigger the download zip.

**Comment by Laurvin** [comment]
I should have added that all I am using is the Quick Start code from the Readme, not the service worker as my files are only a few lines long each (combined into one file it's not even 200 KB).

I am just itterating an unordered list on my page of quotes and want to save those results as separate files and download them all at once in a zip. So I don't need to fetch anything from external url's or do anything else complicated.

It's just a few lines of code:

> const allDownloads = [];
>  
> $( "#quotesList li" ).each(function( index )
> {
>   allDownloads.push({ name: $(this).data("count") + ".txt", lastModified: new Date(), input: $(this).text() })
> });

**Comment by Touffy** [comment]
Yes, in your case, you'll be totally fine with the Blob variant.

You can ignore the lastModified property if it's going to be new Date(), because that's the default value.

So you've already mapped your DOM to an array of suitable input objects. Just showing a version without jQuery, does the same thing as your code :

```js
const quotes = Array.from(document.querySelectorAll("#quotesList li"), li => ({
  input: li.textContent,
  name: li.dataset.count + ".txt"
}))
```

Just make sure every list item matching your selector has a unique "data-count" attribute (otherwise the name of multiple entries may conflict and you'll get a corrupted archive). You could also use the iteration index to be safe.

And now… well, you just call `downloadZip` with it and trigger a download as shown in the quick start !

```js
const blob = await downloadZip(quotes).blob()

const link = document.createElement("a")
link.href = URL.createObjectURL(blob)
link.download = "quotes.zip"
link.click()
link.remove()
URL.revokeObjectURL(link.href)
```

**Comment by Laurvin** [comment]
Thanks, it works great now, even the jQuery version!
